tests/test_disable_signal_mixin.py

```diff
@@ -1,27 +1,80 @@
-from django.db.models.signals import post_save
-from django.test import TestCase
-
+from django.db.models.signals import post_save, post_delete
 from startups.models import Startup
-from startups.signals import update_startup_document
+from startups.signals import update_startup_document, delete_startup_document
+from unittest.mock import patch
 
+# Import all Elasticsearch documents that may be triggered during test data creation
+from startups.documents import StartupDocument
+from projects.documents import ProjectDocument
+# Add other documents here if needed:
+# from investors.documents import InvestorDocument
+# from investments.documents import SubscriptionDocument
 
-class DisableSignalMixin(TestCase):
-    sender = None
+class DisableElasticsearchSignalsMixin:
+    """
+    Mixin to disable Elasticsearch indexing during tests.
+    Disconnects Django signals and mocks update/delete methods of all registered documents.
+    """
 
     @classmethod
-    def disable_signal(cls):
+    def setUpClass(cls):
+        # Disconnect signals to prevent automatic indexing
         post_save.disconnect(update_startup_document, sender=Startup)
+        post_delete.disconnect(delete_startup_document, sender=Startup)
 
-    @classmethod
-    def enable_signal(cls):
-        post_save.connect(update_startup_document, sender=Startup)
+        # Patch update/delete methods of all relevant documents
+        cls.es_patchers = [
+            patch.object(StartupDocument, 'update', lambda self, instance, **kwargs: None),
+            patch.object(StartupDocument, 'delete', lambda self, instance, **kwargs: None),
+            patch.object(ProjectDocument, 'update', lambda self, instance, **kwargs: None),
+            patch.object(ProjectDocument, 'delete', lambda self, instance, **kwargs: None),
+            # Add more patchers here if needed
+        ]
```

Comment:
Patching document methods with lambdas may mask errors if the document API changes. Recommend using unittest.mock.patch context managers and asserting calls.
```
with patch.object(StartupDocument, 'update', return_value=None):
    # test code here
```